internal/pkg/describe/app.go

```diff
@@ -111,26 +113,24 @@ func (d *AppDescriber) Version() (string, error) {
 	}
 	stackMetadata, stackSetMetadata := metadata{}, metadata{}
 
-	appStackName := stack.NameForAppStack(d.app)
-	appStackMetadata, err := d.cfn.Metadata(cloudformation.MetadataWithStackName(appStackName))
+	appStackMetadata, err := d.stackDescriber.StackMetadata()
 	if err != nil {
-		return "", fmt.Errorf("get metadata for app stack %s: %w", appStackName, err)
+		return "", err
 	}
 	if err := yaml.Unmarshal([]byte(appStackMetadata), &stackMetadata); err != nil {
-		return "", fmt.Errorf("unmarshal Metadata property for app stack %s: %w", appStackName, err)
+		return "", fmt.Errorf("unmarshal Metadata property for app stack %s: %w", cfnstack.NameForAppStack(d.app), err)
```

Comment:
nit: can we remove the dependency on `cfnstack`? maybe just 
```suggestion
		return "", fmt.Errorf("unmarshal Metadata property from app %s stack: %w", d.app, err)
```